lib/sm/client.rb

```diff
@@ -108,22 +108,35 @@ def delete_folder(id)
     #
     # @return [Common::Collection]
     #
-    def get_folder_messages(folder_id)
-      page = 1
-      json = { data: [], errors: {}, metadata: {} }
-
-      loop do
-        path = "folder/#{folder_id}/message/page/#{page}/pageSize/#{MHV_MAXIMUM_PER_PAGE}"
-        page_data = perform(:get, path, nil, token_headers).body
-        json[:data].concat(page_data[:data])
-        json[:metadata].merge(page_data[:metadata])
-        break unless page_data[:data].size == MHV_MAXIMUM_PER_PAGE
-
-        page += 1
+    # rubocop:disable Metrics/MethodLength
+    def get_folder_messages(user_uuid, folder_id, use_cache)
+      cache_key = "#{user_uuid}-folder-messages-#{folder_id}"
+      messages = nil
+      messages = Message.get_cached(cache_key) if use_cache
+
+      if messages
+        Rails.logger.info('secure messaging folder messages cache fetch', cache_key)
+        Common::Collection.new(Message, { data: messages })
+      else
+        Rails.logger.info('secure messaging folder messages service fetch', cache_key)
+        page = 1
+        json = { data: [], errors: {}, metadata: {} }
+
+        loop do
+          path = "folder/#{folder_id}/message/page/#{page}/pageSize/#{MHV_MAXIMUM_PER_PAGE}"
+          page_data = perform(:get, path, nil, token_headers).body
+          json[:data].concat(page_data[:data])
+          json[:metadata].merge(page_data[:metadata])
+          break unless page_data[:data].size == MHV_MAXIMUM_PER_PAGE
+
+          page += 1
+        end
+        messages = Common::Collection.new(Message, json)
+        Message.set_cached(cache_key, messages)
+        messages
```

Comment:
Thinking I might also be able to use the `get_cached_or fetch_data` helper method I created below with `yield` somehow, but still getting a grasp of how to do that.